### PR TITLE
Fixing array incides to compute the correct rates of convergence

### PR DESCRIPTION
**Pull request by francispoulin**
The current version is broken and does not compute the correct exponents.

The corrected version fixes this problem.

### CODE REVIEW

**Comment by francispoulin** [comment]
Since this scheme computes the rates of convergence acurately, I wonder if we might want to include this as a test, or validation.  Any thoughts?

**Comment by francispoulin** [comment]
There is an error in `initailize-gpu-environment`, copied below, which I don't believe comes from this change.  Can someone reset the tests?  I don't believe I have the authority to do that.

```
ERROR: Unable to automatically install 'CUDNN' from '/data5/glwagner/.julia-2937/packages/CUDA/mVgLI/Artifacts.toml'
 ```

**Review by tomchor — APPROVED** [review]

File: validation/advection/plot_rates_convergence_advection.jl [hunk 1]
```diff
@@ -39,15 +39,14 @@ halos(::UpwindBiasedFifthOrder) = 3
...
-L  = 2.5
 W  = 0.1
 Ns = 2 .^ (6:10)
 Δt = 0.01 * minimum(L/Ns) / U
 pnorm = 1
 
 c(x, y, z, t, U, W) = exp( - (x - U * t)^2 / W^2 )
    h(x, y, z) = 1
-  uh(x, y, z) = 1 
+  uh(x, y, z) = 1 * U
```
[the remark this answers is not in the document]
Comment (francispoulin) [reply validation/advection/plot_rates_convergence_advection.jl:49]: Very good point.  I will make that change and then merge.  Thanks!

**Comment by francispoulin** [comment]
I see there is a failed test but from looking at the error it seems to be due to exceeding a disk quota.  I will proceed to merge as this code is not actually used in any of the tests.

**Pull request merged**